File: public/blackjack/Firebase/deck.js
// deck.js
const db = admin.firestore();

async function createDeck() {
    console.log("Creating deck...");
    const deckCollection = db.collection("Deck");
  
    const suits = ["Hearts", "Diamonds", "Clubs", "Spades"];
    const ranks = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace"];
  
    const cardPromises = [];
    for (const suit of suits) {
      for (const rank of ranks) {
        const card = {
          suit: suit,
          rank: rank,
          img: `images/${rank.toLowerCase()}_of_${suit.toLowerCase()}.png`,  // Ensure rank is lowercased
        };
        console.log("Adding card:", card);
        cardPromises.push(deckCollection.add(card));
      }
    }
  
    try {
      await Promise.all(cardPromises);
      console.log("Deck created successfully!");
    } catch (error) {
      console.error("Error creating deck:", error);
    }
  }
  
  // Call the function when this file is executed
  createDeck().then(() => {
    console.log("Done!");
    process.exit();
  }).catch((error) => {
    console.error("Error:", error);
    process.exit(1);
  });
  

File: public/blackjack/Java/Deck.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Deck {
    private List<Card> cards; // This is the relationship, Deck contains Cards

    public Deck() {
        this.cards = new ArrayList<>();
        String[] suits = {"Hearts", "Diamonds", "Clubs", "Spades"};
        String[] values = {"2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace"};

        for (String suit : suits) {
            for (String value : values) {
                cards.add(new Card(suit, value)); // Creating Card objects
            }
        }
        shuffle(); // Shuffle the deck upon creation
    }

    public void shuffle() {
        Collections.shuffle(cards);
    }

}
